### PR TITLE
fix: #1116 Dialog dismissed after fetching of data

### LINKED ISSUE #1116: Infinite loader while downloading transaction receipt
**Summary:**
An infinite loader is shown to user while downloading transaction receipt and rejecting the storage permission

**Steps to reproduce:**
1. Deny storage permissions from Phone's setting
2. Open Transfer
2. Open History
3. Click on a transaction
4. Click on 'Download' Floating action button
5. Deny the storage permission
6. Observe

**Expected behavior:**

What did you expect the app to do?
The permission dialog must dismiss

What did you see instead?  Describe your issue in detail here.
Permission dialog dismisses but an infinite loader is visible to user

**Device and Android version:**
OnePlus 6
Android 10

**Screenshots:**

https://user-images.githubusercontent.com/50913976/103731823-12319100-500c-11eb-8f5c-6f9909d658e5.mp4

### PR DESCRIPTION
**Pull request by divyank00**
## Issue Fix
Fixes #1116

## Screenshots

https://user-images.githubusercontent.com/50913976/103733776-7fdfbc00-5010-11eb-9009-4775ec943ec6.mp4

## Description
Made sure dialog is dismissed completely

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by divyank00** [comment]
@devansh-299 @s-ayush2903  Please Review

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ public void showTransferDetail(TransferDetail transferDetail) {
         tvTransFromName.setText(Constants.NAME + transferDetail.getFromClient().getDisplayName());
         tvTransFromNumber.setText(Constants.ACCOUNT_NUMBER + transferDetail
                 .getFromAccount().getAccountNo());
-        hideProgressDialog();
+        cancelProgressDialog();
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java]: Why cancel it? We could have hiden it simply it _after_ the permission granting results

Comment (divyank00) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java]: Due to hiding the dialog didnt become null, so after the permission dialog the dialog was recreated repeatedly

Comment (devansh-299) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java]: Sorry, I meant dismissing it. Canceling also makes a call to `onCancelListener` interface if I remember correctly. Simply dismissing after it's work is done seems better to me

Comment (divyank00) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java]: Yeah will dismiss it then and check if it still works

**Comment by divyank00** [comment]
@devansh-299 have changed the code. Take a look

**Pull request merged**